core/src/main/java/org/keycloak/representations/account/UserProfileMetadata.java

```diff
@@ -42,4 +45,13 @@ public void setAttributes(List<UserProfileAttributeMetadata> attributes) {
         this.attributes = attributes;
     }
 
+    public UserProfileAttributeMetadata getAttributeMetadata(String attributeName) {
+        return Optional.ofNullable(getAttributes()).orElse(Collections.emptyList()).stream().filter(
+                new Predicate<UserProfileAttributeMetadata>() {
+                    @Override
+                    public boolean test(UserProfileAttributeMetadata metadata) {
+                        return metadata.getName().equals(attributeName);
+                    }
+                }).findAny().orElse(null);
+    }
```

Comment:
See this also brings me to a question. Do we really need for the attributes metadata to be a list? Even from the API perspective (that will be consumed by the UI) it would IMHO made more sense for it to be a map with attribute names as the keys. Anyway, not blocking for me.

Reply:
Attributes are ordered and a list fits better when returning from over REST.